crates/provider-wadm/src/lib.rs

```diff
@@ -0,0 +1,583 @@
+                                    warn!("Received error status: {}", status_response.message);
+                                }
+                                StatusResult::NotFound => {
+                                    warn!("Status not found for: {}", app_name.clone());
+                                }
+                                StatusResult::Ok => {
+                                    if let Some(status) = status_response.status {
+                                        debug!(?status, ?component_id, "received status");
+
+                                        let span =
+                                            tracing::debug_span!("handle_message", ?component_id);
+                                        let permit = match semaphore.clone().acquire_owned().await {
+                                            Ok(p) => p,
+                                            Err(_) => {
+                                                warn!("Work pool has been closed, exiting queue subscribe");
+                                                break;
+                                            }
+                                        };
+
+                                        let component_id = Arc::clone(&component_id);
+                                        let app_name = Arc::clone(&app_name);
+                                        let wrpc = Arc::clone(&wrpc);
+                                        tokio::spawn(async move {
+                                            dispatch_status_update(
+                                                &wrpc,
+                                                component_id.as_str(),
+                                                &app_name,
+                                                status.into(),
+                                                permit,
+                                            )
+                                            .instrument(span)
+                                            .await;
+                                        });
+                                    } else {
+                                        warn!("Received status OK but no status provided");
+                                    }
+                                }
+                            },
+                            Err(e) => {
+                                warn!("Failed to deserialize message: {}", e);
+                            }
+                        };
+                    }
+                }
+            },
+            abort_registration,
+        ));
+
+        Ok(abort_handle)
+    }
+
+    /// Helper function to get the NATS client from the context
+    async fn get_client(&self, ctx: Option<Context>) -> anyhow::Result<Client> {
+        if let Some(ref source_id) = ctx
+            .as_ref()
+            .and_then(|Context { component, .. }| component.clone())
+        {
+            let actors = self.consumer_components.read().await;
```

Comment:
This is probably a holdover from the old code before reopening, but this should be `components` instead of `actor`